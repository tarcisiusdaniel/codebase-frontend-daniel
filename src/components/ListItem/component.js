import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

export default class Component extends React.Component {
  render() {
    const { classes, text, url } = this.props;
    return (
      <Typography>
        <Link className={classes.item} to={url}>
          {text}
        </Link>
      </Typography>
    );
  }
}

Component.defaultProps = {
  classes: {},
  text: '',
  url: '',
};

Component.propTypes = {
  classes: PropTypes.object,
  text: PropTypes.string,
  url: PropTypes.string
};