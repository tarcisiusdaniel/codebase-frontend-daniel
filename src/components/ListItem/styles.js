import { COLOR_PRIMARY, COLOR_PRIMARY_2, COLOR_WHITE } from '../../styles';

const styles = {
  item: {
    '&:hover': {
      backgroundColor: COLOR_PRIMARY_2
    },
    backgroundColor: COLOR_PRIMARY,
    borderRadius: '10px',
    color: COLOR_WHITE,
    margin: '0 10px',
    padding: '2px 15px',
    textDecoration: 'none',
  }
};

export default styles;