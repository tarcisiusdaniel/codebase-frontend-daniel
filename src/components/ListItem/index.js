import { withStyles } from '@material-ui/core/styles';
import Component from './component';
import styles from './styles';

export default withStyles(styles)(Component);