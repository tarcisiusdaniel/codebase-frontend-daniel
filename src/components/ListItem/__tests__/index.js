import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../index';

jest.mock('react-router-dom');

describe('ListItem', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ListItem text="Hello" url="localhost" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});