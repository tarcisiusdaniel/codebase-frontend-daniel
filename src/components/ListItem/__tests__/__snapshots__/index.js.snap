// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<p
  className="MuiTypography-root-2 MuiTypography-body1-11"
/>
`;
